src/_pytest/config/__init__.py

```diff
@@ -1525,6 +1525,8 @@ def _getini(self, name: str):
                     return default
                 if type is None:
                     return ""
+                if type == "string":
```

Comment:
im very torn on this change in fear of the butterfly effec
at the same time and more propper solution will need a new api to avoid hell for users
@nicoddemus whats your impression on this

in particular the detail about the type is such, that for both None and string, it also makes sense to return  a empty string